### DISCUSSION TITLE
Using Deepgram Nova-2 with MP4 vs WAV file

### DISCUSSION THREAD

**Discussion by UtsavChokshiCNU** [deepgram/community#492] [General help] [answered]
Hello Everyone,

I have some questions around best practices for getting highest quality transcripts.

**Q1. :** We want to transcribe a single person video. 

**Option #1 :** 
Should we use `Nova-2` by directly sending `mp4` video file 

**Option #2 :** 
Should we convert video into `wav` file and then use it ?

**Problem with option #2 is** : we use ffmpeg to convert video to audio. Sometimes it happens that wav file generated  is longer (few seconds) than original mp4 video file. Here is ffmpeg command for reference : 
`ffmpeg -i x.mp4 -acodec pcm_s16le -ar 44100 -ac 2 x.wav`

**Q2 :** Should we divide original MP4 or wav file into chunks and then submit it to Deepgram ? 
Because right now, we are submitting complete file (which can be around 2 hours) and expecting transcript out of it. It works perfectly fine.

```
 # Set the source
    source = {
      'buffer': audio,
      'mimetype': MIMETYPE
    }

  # Send the audio to Deepgram and get the response
  response = await asyncio.create_task(
    deepgram.transcription.prerecorded(
      source,
      {
        'smart_format': True,
        'model': 'nova-2',
        'utterances': True
      }
    )
  )
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment, marked as answer]
Hey @UtsavChokshiCNU, the best practice is to extract the audio from the mp4 file - just like you're doing. A 2 hour video file may be very large and could run up against our 2GB limit. Larger files also take much longer to process because, while Deepgram will process the audio in under a minute, the file upload time can take much longer.

It's strange that the .wav file is a few seconds longer than the video. That indicates the metadata (e.g. sample rate) for the mp4 file is slightly off. Are the last few seconds of the file silent, or does the audio audio get slowed down slightly to fit in a longer duration? If the latter, it's likely a mismatch between the mp4's sample rate and the .wav's sample rate.

Q2: No need to break the file up! In fact, unless the file is over the 2 GB upload limit or you need to perform complex language detection, we encourage you to send the full file.
